### ISSUE TITLE
File conflicts when installing Warewulf Node Health Check (NHC) tool

### ISSUE THREAD

**Issue by eshelman** [openhpc/ohpc#135]
I'm afraid there's trouble when trying to install the OpenHPC build of the NHC tool. Here's what I'm getting on a fresh install attempt:

```
[root@head ~]# yum -y install --installroot /opt/ohpc/admin/images/centos-7/ warewulf-nhc-ohpc
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.netnitco.net
 * epel: mirror.cogentco.com
 * extras: mirror.vcu.edu
 * updates: mirror.netdepot.com
Resolving Dependencies
--> Running transaction check
---> Package warewulf-nhc-ohpc.noarch 0:1.4.1-2.1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

====================================================================================================================================================================================================================
 Package                                                  Arch                                          Version                                            Repository                                          Size
====================================================================================================================================================================================================================
Installing:
 warewulf-nhc-ohpc                                        noarch                                        1.4.1-2.1                                          OpenHPC_1.0                                         61 k

Transaction Summary
====================================================================================================================================================================================================================
Install  1 Package

Total size: 61 k
Installed size: 252 k
Downloading packages:
Running transaction check
Running transaction test


Transaction check error:
  file /opt/ohpc/pub from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package slurm-ohpc-15.08.1-4.1.x86_64
  file /opt/ohpc/pub/doc from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package slurm-ohpc-15.08.1-4.1.x86_64
  file /opt/ohpc/pub/doc/contrib from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package slurm-ohpc-15.08.1-4.1.x86_64
  file /opt/ohpc/pub from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package lmod-ohpc-6.0.10-2.2.x86_64
  file /opt/ohpc/pub/doc from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package lmod-ohpc-6.0.10-2.2.x86_64
  file /opt/ohpc/pub/doc/contrib from install of warewulf-nhc-ohpc-1.4.1-2.1.noarch conflicts with file from package lmod-ohpc-6.0.10-2.2.x86_64

Error Summary
-------------

```

**Comment by JohnWestlund** [comment]
Good catch. I’ll commit a fix internally shortly and we’ll have the updated RPMs available when the updates repo goes live.

**Comment by koomie** [comment]
@crbaird, warewulf-nhc has been added to the 1.0.1 project area in OBS. Can you update to resolve the file conflict error?

**Comment by crbaird** [comment]
installing from Updates now succeeds

**Comment by crbaird** [comment]
Hi @eshelman, we think we've gotten our update strategy sorted out and warewulf-nhc should install correctly from there. Though the 1.0.1 release isn't official yet, you can test it by enabling the following repo:

http://build.openhpc.community/OpenHPC:/1.0:/Update1:/Factory/CentOS_7.1/OpenHPC:1.0:Update1:Factory.repo

**Comment by eshelman** [comment]
Hi Folks,

This bug found it's way back into v1.1 and v1.1.1 (though the fix does appear to be in the v1.0.1 branch).

Thanks!

**Comment by JohnWestlund** [comment]
The issue is that the nhc rpm changes the umask before the make install and
doesn't set it back. So directories in this rpm have different perms then
those in just about every other OHPC rpm.

A umask 0022 after the make install will resolve the problem.

John
